### ISSUE TITLE
🐛 cannot extend configurations that extend other configurations (2+ nesting)

### ISSUE THREAD

**Issue by nevir** [biomejs/biome#1867]
### Environment information

```block
CLI:
  Version:                      1.5.3
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v20.11.0"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "yarn/4.1.0"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 true

Workspace:
  Open Documents:               0
```

### What happened?

See https://codesandbox.io/p/live/1c1835c5-c10f-4590-832f-51a122b1588a

TL;DR: If you have a chain of configuration files more than one level deep (biome.json -> parent.json -> grandparent.json), subsequent configuration (grandparent.json or deeper) is not included by extension. 

It appears as if subsequent `extends` properties are silently ignored.

### Expected result

Expected to be able to extend configuration at any depth of extension (for example: more complex shared configurations & repository layouts)

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by ematipico** [comment]
The feature was designed for just one level. In fact we don't document a "grandparent" example.

Feel free to open a discussion to propose this use case

**Comment by PaulRBerg** [comment]
Created a discussion to suggest this as a feature: https://github.com/biomejs/biome/discussions/8257
